src/Couchbase.Lite.Shared/Sync/WebSocketWrapper.cs

```diff
@@ -198,20 +198,33 @@ public void Start()
 
                 // STEP 2: Open the socket connection to the remote host
                 var cts = new CancellationTokenSource();
-                cts.CancelAfter(ConnectTimeout);
-                try {
+                var tok = cts.Token;
+                
+                tok.Register(() =>
+                {
+                    _client.Close();
+                });
+
+                try
+                {
                     _client.ConnectAsync(_logic.UrlRequest.Host, _logic.UrlRequest.Port).ContinueWith(t =>
                     {
                         if (!NetworkTaskSuccessful(t)) {
                             return;
                         }
 
                         _queue.DispatchAsync(StartInternal);
-                    }, cts.Token);
-                } catch (Exception e) {
+
+                        if (tok.IsCancellationRequested)
+                            return;
```

Comment:
The above two lines are unneeded.  Actually it is not possible to make it this far with a cancelled token since it will return false from `NetworkTaskSuccessful`